engines/director/lingo/lingo-object.cpp

```diff
@@ -508,6 +508,22 @@ Datum ScriptContext::getProp(const Common::String &propName) {
 	return _properties[propName]; // return new property
 }
 
+Common::String ScriptContext::getPropAt(uint32 index) {
+	// FIXME: Refactor the whole thing to have ordered keys
```

Comment:
See `Common::StableMap`

Reply:
The key order seems to be the order they are added to the context, so to keep things simple I've added another array to keep track of that.